### DISCUSSION TITLE
Qtile doesn't start and can't load the default config

### DISCUSSION THREAD

**Discussion by greg086** [qtile/qtile#5099] [Q&A] [answered]
I have ran into issues starting qtile. It kicks me back to the sddm login screen. I can't load the default config file so it's nothing with my config. Qtile won't even copy from the location in /usr/share/ if I delete the config in ~/.config/qtile.

The problem did start shortly after a system upgrade although I remember qtile working right after upgrading. The only thing I'm seeing from the log file is a dependency error (last line) but it won't tell me what dependency. The error has only shown up on the 17th when the problem started and qtile has not written to the log since even though I've tried to start it multiple times when troubleshooting. I've tried downgrading to a previous qtile but I still can't get back in.

OS: Arco Linux

[qtile.log](https://github.com/user-attachments/files/17912518/qtile.log)

**Comment by tych0** [comment]
```
2024-04-22 20:42:00,476 ERROR libqtile manager.py:restart():L251 Preventing restart because of a configuration error:
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 249, in restart
    self.config.load()
  File "/usr/lib/python3.11/site-packages/libqtile/confreader.py", line 131, in load
    self._reload_config_submodules(path)
  File "/usr/lib/python3.11/site-packages/libqtile/confreader.py", line 120, in _reload_config_submodules
    importlib.reload(module)
  File "/usr/lib/python3.11/importlib/__init__.py", line 169, in reload
    _bootstrap._exec(spec, module)
  File "<frozen importlib._bootstrap>", line 621, in _exec
  File "<frozen importlib._bootstrap_external>", line 936, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1074, in get_code
  File "<frozen importlib._bootstrap_external>", line 1004, in source_to_code
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/greg/.config/qtile/theme.py", line 168
    **powerline
    ^^
SyntaxError: invalid syntax
```

though it is unfortunately muddied by some other benign errors that #5020 should fix, should I ever get around to debugging and landing it.

**Reply by elParaguayo** [reply]
`**powerline` is a `qtile-extras` thing so definitely not the default config.

**Reply by greg086** [reply]
> ```
> 2024-04-22 20:42:00,476 ERROR libqtile manager.py:restart():L251 Preventing restart because of a configuration error:
> Traceback (most recent call last):
>   File "/usr/lib/python3.11/site-packages/libqtile/core/manager.py", line 249, in restart
>     self.config.load()
>   File "/usr/lib/python3.11/site-packages/libqtile/confreader.py", line 131, in load
>     self._reload_config_submodules(path)
>   File "/usr/lib/python3.11/site-packages/libqtile/confreader.py", line 120, in _reload_config_submodules
>     importlib.reload(module)
>   File "/usr/lib/python3.11/importlib/__init__.py", line 169, in reload
>     _bootstrap._exec(spec, module)
>   File "<frozen importlib._bootstrap>", line 621, in _exec
>   File "<frozen importlib._bootstrap_external>", line 936, in exec_module
>   File "<frozen importlib._bootstrap_external>", line 1074, in get_code
>   File "<frozen importlib._bootstrap_external>", line 1004, in source_to_code
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "/home/greg/.config/qtile/theme.py", line 168
>     **powerline
>     ^^
> SyntaxError: invalid syntax
> ```
> 
> though it is unfortunately muddied by some other benign errors that #5020 should fix, should I ever get around to debugging and landing it.

the date on that error is  2024-04-22, back in April. That's something I was able to fix. The current problem has been happening since the 17th of this month.

**Comment by greg086** [comment]
Update, the problem seams to be something with my home folder.

I tried installing Arco on a different partition and setting my /home to the drive that I use for my `/home` directory. Qtile won't launch. When I tried again and let `/home` be inside `/`, I was able to get into qtile and my regular configs worked.

I'm not sure where to go from here.

**Comment by tych0** [comment]
We probably need ~/.xsession-errors here, I don't see anything obvious in your qtile log.

**Reply by greg086** [reply]
~/.xsession-errors is apparently a blank file on my system.

**Reply by tych0** [reply]
Can you share your .xsession, or however you are invoking qtile?

**Reply by greg086** [reply]
It's through sddm with the standard .desktop file
```
[Desktop Entry]
Name=Qtile
Comment=Qtile Session
Exec=qtile start
Type=Application
Keywords=wm;tiling
```

**Comment by greg086** [comment, marked as answer]
Problem is fixed now. Like I said earlier the problem was with my /home directory. I reinstalled linux with a fresh home and mounted the drive where my old /home was for documents and such.

This had nothing to do with qtile itself as i3 and awesome weren't working either. Thanks for the help anyway.

**Reply by tych0** [reply]
My guess is that something was wonky with actually starting qtile, probably your display manager has some logs about failing to start it, maybe it needed some PYTHONPATH hackery or similar. Anyway, glad you got it working.
